group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/GroupSpecImpl.java

```diff
@@ -18,48 +18,60 @@
 
 import org.apache.kafka.common.Uuid;
 
+import java.util.Collection;
+import java.util.Collections;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
 
 /**
  * The assignment specification for a consumer group.
  */
 public class GroupSpecImpl implements GroupSpec {
     /**
-     * The member metadata keyed by member Id.
+     * Member subscription metadata keyed by member Id.
      */
-    private final Map<String, AssignmentMemberSpec> members;
+    private final Map<String, MemberSubscriptionSpec> memberSubscriptions;
 
     /**
-     * The subscription type followed by the group.
+     * The subscription type of the group.
      */
     private final SubscriptionType subscriptionType;
 
+    /**
+     * Partitions currently assigned to each member keyed by topicId.
+     */
+    private final Map<String, Map<Uuid, Set<Integer>>> currentAssignment;
```

Comment:
nit: Should we use `assignment` and `invertedAssignment`?

Reply:
I used memberAssigment and invertedMemberAssignment